src/azul/service/avro_pfb.py

```diff
@@ -143,7 +153,10 @@ def from_json(cls,
         # For files, document_id is not unique (because of related_files), but
         # uuid is.
         ids = [object_['uuid']] if name == 'files' else sorted(object_['document_id'])
-        return cls(id='_'.join((name, *ids)), name=name, object=object_)
+        joiner = '_'
+        reject(any(joiner in id_ for id_ in ids), ids)
+        id_ = f'{name}_{uuid5(cls.namespace_uuid, joiner.join(ids))}_{len(ids)}'
```

Comment:
Assert that `name` does not contain underscore.

Reply:
@hannes-ucsc: "Since name may contain an underscore, use `:` or `/` for the second joiner. Also use .join instead of format string."